Entities/Characters/Scripts/UnSpawnImmunity.as

```diff
@@ -33,11 +39,16 @@ void onTick(CBlob@ this)
 
 	if (!immunity || this.getPlayer() is null)
 	{
-		this.Untag("invincible");
-		this.Tag("invincibility done");
-		this.Sync("invincibility done", true);
-
-		this.getCurrentScript().runFlags |= Script::remove_after_this;
-		this.getSprite().setRenderStyle(RenderStyle::normal);
+		EndInvincibility(this);
 	}
 }
+
+void EndInvincibility(CBlob@ this)
+{
+	this.Untag("invincible");
+	this.Tag("invincibility done");
+	this.Sync("invincibility done", true);
```

Comment:
Somewhat wasteful as this will end up syncing this property every time ontick

Reply:
I just moved the existing code. I'm not sure we really need this tag in the first place, there might be a way to replace it with a different check.